runtime/src/main/kotlin/kotlin/Annotations.kt

```diff
@@ -148,4 +149,23 @@ public annotation class DslMarker
 @Retention(AnnotationRetention.BINARY)
 @MustBeDocumented
 @SinceKotlin("1.1")
-public annotation class PublishedApi
\ No newline at end of file
+public annotation class PublishedApi
+
+/**
+ * This annotation indicates what exceptions should be declared by a function when compiled to a platform method.
+ *
+ * When compiling to Objective-C/Swift framework, functions having or inheriting
+ * this annotation are represented as `NSError*`-producing methods in Objective-C
+ * and as `throws` methods in Swift.
+ *
+ * When Kotlin function called from Swift/Objective-C code throws an exception
+ * which is an instance of one of the [exceptionClasses] or their subclasses,
+ * it is propagated as `NSError`. Other Kotlin exceptions reaching Swift/Objective-C
+ * are considered unhandled and cause program termination.
+ *
+ * @property exceptionClasses the list of checked exception classes that may be thrown by the function.
+ */
+@SinceKotlin("1.4")
+@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER, AnnotationTarget.CONSTRUCTOR)
```

Comment:
I've extended targets list to match JVM annotation. Hope that wouldn't be a problem.

Reply:
I'm not sure backend is ready for this.
I guess `@Throws` should be forbidden on property accessors for native in frontend.

Reply:
Reverted accessor targets for K/N.